tiledb/sm/subarray/subarray.h

```diff
@@ -1026,6 +1130,41 @@ class Subarray {
     uint64_t range_len_;
   };
 
+  /**
+   * Wrapper for optional<tuple<std::string, RangeSetAndSuperset>> for
+   * cleaner data access.
+   */
+  struct LabelRangeSubset {
+   public:
+    LabelRangeSubset();
+
+    LabelRangeSubset(
+        const DimensionLabelReference& ref, bool coalesce_ranges = true);
+
+    inline bool has_value() const {
```

Comment:
I added the basic doxygen docs. I don't think I can remove the copy/move operations since they are used to create a vector of LabelRangeSubset (needs to at least be copy constructible and copy assignable) and the swap/copy/move for the Subarray class.